### PR TITLE
Added Expression module

### PR DESCRIPTION
**Pull request by salterm**
Mostly changes / additions to the Expression module, but also some changes to the Parser (notably changing its output to (Predicate, Predicate)). Also added basic example file for debugging.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -84,7 +85,14 @@ fn expand_condition_fn(meta: &MetaItem) {
             // FIXME: arguments should be parsed by the parser module, not in this control module
             // NOTE: EXPERIMENT: control flow happens here
             let mut builder = Attr {pre: None, post: None, pre_str: "".to_string(), post_str: "".to_string()};
-            parser::expand_args(&mut builder, args);
+
+            // Parse the conditions
+            let conditions = parser::expand_args(&mut builder, args);
```
Comment (found101) [inline src/lib.rs]: Create separate function instead of tacking onto expand_args, e.g. get_args?

File: src/parser/mod.rs [hunk 2]
```diff
@@ -30,7 +31,7 @@ pub fn demo() {
 
 // FIXME: this should be in the parser module!
 // Parse the condition arguments
-pub fn expand_args(builder: &mut super::Attr, args: &Vec<P<MetaItem>>) {
+pub fn expand_args(builder: &mut super::Attr, args: &Vec<P<MetaItem>>) -> (Predicate, Predicate) {
```
Comment (found101) [inline src/parser/mod.rs]: Create separate function instead of tacking onto expand_args, e.g. get_args?

**Comment by found101** [comment]
As per comments

**Comment by salterm** [comment]
I'm not sure what just happened here. Did the pull request get denied? What do your comments mean?

**Pull request closed without merging**